libnetwork/drivers/overlay/ovmanager/ovmanager.go

```diff
@@ -160,38 +159,26 @@ func (d *driver) NetworkFree(id string) error {
 }
 
 func (n *network) obtainVxlanID(s *subnet) error {
-	var (
-		err error
-		vni uint64
-	)
-
-	n.mu.Lock()
-	vni = uint64(s.vni)
-	n.mu.Unlock()
-
+	vni := uint64(s.vni)
 	if vni == 0 {
-		vni, err = n.driver.vxlanIdm.GetIDInRange(vxlanIDStart, vxlanIDEnd, true)
+		vni, err := n.driver.vxlanIdm.GetIDInRange(vxlanIDStart, vxlanIDEnd, true)
```

Comment:
This chunk is slightly confusing, as it looks like we're deliberately creating a new `vni` variable, but with the same name as outside the block. The code is _correct_, but wondering if it would be more clear to either use a different variable name, or perhaps we should reverse the conditions;

```go
func (n *network) obtainVxlanID(s *subnet) error {
	if vni := uint64(s.vni); vni != 0 {
		return n.driver.vxlanIdm.GetSpecificID(vni)
	}

	vni, err := n.driver.vxlanIdm.GetIDInRange(vxlanIDStart, vxlanIDEnd, true)
	if err != nil {
		return err
	}

	s.vni = uint32(vni)
	return nil
}
```

Very much side-note; it's a bit awkward that we have to do the uint64 <--> uint32 shuffle here 😞

Reply:
DOH! The next commit removes this chunk, so guess there's no need to update the intermediate commit